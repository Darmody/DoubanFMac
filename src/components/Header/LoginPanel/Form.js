// @flow
import React, { PureComponent } from 'react'
import styled from 'styled-components'
import { Field } from 'redux-form'

const LoginForm = styled.form`
  width: 22.375rem;
  padding: 3.75rem 3.75rem 3.125rem 3.75rem;
`
const Title = styled.h2`
  text-align: center;
  font-size: 1.563rem;
  font-weight: normal;
  margin-bottom: 1.875rem;
  line-height: 1rem;
`

const FieldWrapper = styled.div`
  height: 2.5rem;
  margin-top: .875rem;
  overflow: hidden;
`

type InputProps = {
  className: string,
  input: Object,
  placeholder: string,
  type: string,
}
const ReduxInput = ({
  input,
  type = 'text',
  placeholder = '',
  className = '',
}: InputProps) => (
  <input type={type} placeholder={placeholder} {...input} className={className} />
)
const Input = styled(ReduxInput)`
  border: 1px solid #ccc;
  width: calc(100% - .75rem);
  height: .875rem;
  line-height: .875rem;
  padding: .75rem 0 .75rem .625rem;
  color: #111;
  font-size: .813rem;
  outline: none;
  cursor: text;
  border-radius: 3px;
  box-shadow: inset 0 0 4px rgba(0,0,0,0.06);
`

const Button = styled.input`
  display: block;
  border: none;
  width: 100%;
  height: 2.5rem;
  border-radius: 3px;
  background-color: #59b36a;
  color: white;
  cursor: pointer;
`

type Props = {
  handleSubmit: Function,
}

export default class LoginFormComponent extends PureComponent {
  props: Props

  render() {
    return (
      <LoginForm>
        <Title>欢迎来到豆瓣, 请登录</Title>
        <FieldWrapper>
          <Field
            name="username"
            type="text"
            placeholder="手机号/邮箱/用户名"
            component={Input}
          />
        </FieldWrapper>
        <FieldWrapper>
          <Field
            name="password"
            type="password"
            placeholder="密码"
            component={Input}
          />
        </FieldWrapper>
        <FieldWrapper>
          <Button type="submit" value="登录豆瓣" onClick={this.props.handleSubmit} />
        </FieldWrapper>
      </LoginForm>
    )
  }
}
