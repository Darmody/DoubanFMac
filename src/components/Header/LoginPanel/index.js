// @flow
import React, { PureComponent } from 'react'
import styled from 'styled-components'
import Modal from 'react-modal'
import { LoginForm as LoginFormContainer } from 'containers'
import Form from './Form'

const LoginPanel = styled.a`
  font-size: .75rem;
  margin-left: .625rem;
  line-height: 1.5rem;
  color: #8f8e94;
  text-decoration: none;
  outline: none;
  cursor: pointer;
}
`

const modalStyle = {
  content: {
    top: '50%',
    left: '50%',
    right: 'auto',
    bottom: 'auto',
    marginRight: '-50%',
    transform: 'translate(-50%, -50%)',
  }
}

type Props = {
}

type State = {
  modalIsOpen: boolean,
}

export default class LoginPanelContainer extends PureComponent {
  constructor() {
    super()

    this.state = {
      modalIsOpen: false,
    }
  }

  state: State
  props: Props

  openModal = () => {
    this.setState({ modalIsOpen: true })
  }

  closeModal = () => {
    this.setState({ modalIsOpen: false })
  }

  render() {
    return (
      <div>
        <Modal
          isOpen={this.state.modalIsOpen}
          onRequestClose={this.closeModal}
          contentLabel="登录"
          style={modalStyle}
        >
          <LoginFormContainer component={Form} closeModal={this.closeModal} />
        </Modal>
        <LoginPanel
          href="javascript:void(0);"
          onClick={this.openModal}
        >
          登录
        </LoginPanel>
      </div>
    )
  }
}
