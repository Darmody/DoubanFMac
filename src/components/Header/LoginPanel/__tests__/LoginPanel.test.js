import React from 'react'
import renderer from 'react-test-renderer'
import LoginPanel from 'components/Header/LoginPanel'

describe('Component::LoginPanel', () => {
  test('渲染正确的 DOM 结构', () => {
    const component = renderer.create(<LoginPanel />)
    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })
})
