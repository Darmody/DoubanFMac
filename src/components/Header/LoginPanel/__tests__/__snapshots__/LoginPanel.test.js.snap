// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component::LoginPanel 渲染正确的 DOM 结构 1`] = `
<div>
  <noscript />
  <a
    className="fYxbAV"
    href="javascript:void(0);"
    onClick={[Function]}
  >
    登录
  </a>
</div>
`;
